textpattern/vendors/Textpattern/Tag/Registry.php

```diff
@@ -74,11 +74,9 @@ public function register($callback, $tag = null)
      * @return string|null The tag's results
      */
 
-    public function process($tag, array $atts = null, $thing = null)
+    public function process($tag, array $atts = array(), $thing = null)
     {
-        if ($this->isRegistered($tag)) {
```

Comment:
If you want to remove the second test, then you should throw an exception in ```Textpattern_Tag_Regisry::process()``` and do the check in it.

The function shouldn't generate unexpected, non-catchable errors when an invalid tag name is passed to it. Instead, use exceptions if you want to eliminate the second check, please.